pandas/indexes/numeric.py

```diff
@@ -19,6 +19,32 @@ class NumericIndex(Index):
     """
     _is_numeric_dtype = True
 
+    def __new__(cls, data=None, dtype=None, copy=False, name=None,
```

Comment:
I dont' think we need to accept ``kwargs`` here (or if we do, then use use something like ``core.generic._validate_kwargs`` to allow certain compat keywords (but then we don't advertise them)

(note the location/name of this is changing, but ok for now)